### PR TITLE
fixed issue 509: added getBundle() to AbstractBundleController

### PR DESCRIPTION
**Pull request by mickaelandrieu**
REFS #509

### CODE REVIEW

File: Bundle/AbstractBundleController.php [hunk 1]
```diff
@@ -65,6 +66,24 @@ public function __construct(BBApplication $app)
...
     /**
+     * Get the current bundle
+     *
+     * @param BundleInterface|BundleConfigurationException $bundle if bundle was not set,
+     * return an Exception
+     */
+    public function getBundle()
+    {
+        if (! $this->bundle instanceof BundleInterface) {
+            throw new BundleConfigurationException(
```
Comment (ndufreche) [inline Bundle/AbstractBundleController.php]: Don't throw error if bundle isn't setted, because isn't required.

Comment (mickaelandrieu) [inline Bundle/AbstractBundleController.php]: so ``getBundle`` returns an BundleInterface or null, is that the correct behavior ?

Comment (ndufreche) [inline Bundle/AbstractBundleController.php]: Yes. @eric-chau can validate this behavior ?

Comment (eric-chau) [inline Bundle/AbstractBundleController.php]: I agree with what @ndufreche said especially cause we set the bundle by setter and not the class constructor. So it sounds logical to return ``null`` if not setted.

Comment (eric-chau) [inline Bundle/AbstractBundleController.php]: But I don't really get why we need to retrieve a bundle by controller. Is it really useful?

Comment (ndufreche) [inline Bundle/AbstractBundleController.php]: This is not a system to retrieve a bundle, is just an accessors to help the developer and this only if is automatically setted (currently it work only in the case of an administration bundle controller).

**Comment by mickaelandrieu** [comment]
Such big pull request \o/

**Comment by eric-chau** [comment]
This issue must not exist and it's two distincts issues. First AbstractBundleController has its bundle property as protected so you can access to it from every class that extended it. And the fact that ``::getBundle()`` result on an infinite loop must not be solved by adding the method.

**Comment by ndufreche** [comment]
Yes this resolve just an part of the issue.

**Comment by mickaelandrieu** [comment]
@eric-chau  infinite loop come from __call, if you have getBundle action, this will return null

**Comment by ndufreche** [comment]
Yes but if you called another method non existent, you can reproduce the infinity loop. We have to upgrade __call method to improve it.

**Comment by eric-chau** [comment]
Yeah but this is not a solution to add a getter. Else if you do ``getFooBar`` it will result on an infinite loop too, so to solve this you will create the getter ``getFooBar``? What I mean is that we need to work on ``::__call`` method and not to add the getter. If this getter does not exist before it's because we don't need it.

**Comment by eric-chau** [comment]
Also, I can't find the ``::_call`` method from ``BackBee\Bundle\AbstractBundleController`` or ``BackBee\Controller\Controller`` method, how can @cedric-b get this error? I think he does not extends our controller. This is definitely not BackBee duty to fix this.

**Comment by mickaelandrieu** [comment]
@eric-chau  https://github.com/mickaelandrieu/BackBee/blob/fix-509/Bundle/AbstractBundleController.php#L92

**Comment by eric-chau** [comment]
@mickaelandrieu thank you I didn't notice it.

**Comment by eric-chau** [comment]
So work must be done on this ``::__call`` method and not to add the getter method.

**Comment by ndufreche** [comment]
We need the getter method and the improvement of the __call method.

This PR is correct and ended

**Comment by mickaelandrieu** [comment]
@ndufreche what do I need to do in __call() ?

**Comment by ndufreche** [comment]
Avoid infinite loop and for that rethink the `checkMethodExist` method to fix this problem like check how many action is in the name of the method tested.

**Comment by eric-chau** [comment]
I looked at the ``::__call`` method and this is clearly incomplete. And this method is not useful at all cause we need the bundle in subclass of AbstractBundleController. So why don't you just use ``$this->bundle`` instead of using ``$this->getBundle()``? Plus this will do an useless function call.

**Comment by ndufreche** [comment]
@eric-chau because for implementation, I think is more comprehensive to use an accessors than directly the parameter. Perhaps, in the backbee code we can remove the getter and use the parameter, this is not problematic, but I will keep getBundle method in all cases.

**Comment by eric-chau** [comment]
Ok but to me it's not useful. Nobody will get the bundle from outside of the controller so it will be an unused public method. And inside a subclass it's not optimized to use a getter instead of accessing directly to object property.

**Comment by ndufreche** [comment]
:+1:

**Pull request merged**